scripts/testing/cli.py

```diff
@@ -0,0 +1,603 @@
+
+
+def create_vm(name, vm_type, vm_zone):
+    """
+    Creates a new VM and waits until the startup script finishes its work
+    """
+
+    with click.progressbar(label="Creating a new VM...", length=100) as steps:
+        cmd = [
+            "gcloud",
+            "compute",
+            "instances",
+            "create",
+            name,
+            "--async",
+            "--source-machine-image",
+            SOURCE_MACHINE_IMAGE,
+            "--image-family",
+            IMAGE_FAMILY,
+            "--machine-type",
+            vm_type,
+            "--zone",
+            vm_zone,
+            f"--metadata-from-file={VM_STARTUP_SCRIPT_PATH}",
+        ]
+        result = run_gcloud_cmd(cmd)
+
+        steps.update(25)
+
+        attempts = 0
+        while True:
+            if attempts >= VM_INIT_ATTEMPTS:
+                click.echo("Timeout!")
+                raise click.Abort()
+            attempts = attempts + 1
+
+            # Tries to ssh to the VM and check if the startup script finished its work
+            cmd = [
+                "gcloud",
+                "compute",
+                "ssh",
+                name,
+                "--zone",
+                vm_zone,
+                "--ssh-flag=",
+                "-q",
+                "--command",
+                "sudo ls /var/log/startup-is-finished ",
+            ]
+            result = subprocess.run(cmd, stdout=PIPE, stderr=STDOUT)  # noqa: S603
+
+            stdout = result.stdout.decode("utf-8")
+            # indicates that the VM is booting
+            if re.search("Connection refused", stdout) or re.search(
+                "SSH connectivity issues", stdout
+            ):
+                steps.update(25)
+            # indicates that the VM is ready but the startup script
+            elif result.returncode == 1 or result.returncode == 2:
+                pass
+            elif result.returncode == 0:
+                steps.update(25)
+                break
+            else:
+                click.echo(stdout)
+                raise click.Abort()
+
+            time.sleep(SLEEP_TIMEOUT)
+
+        # update the VM metadata
+        cmd = [
+            "gcloud",
+            "compute",
+            "instances",
+            "add-metadata",
+            name,
+            "--zone",
+            vm_zone,
+            "--metadata",
+            f"division={DIVISION},org={ORG},team={TEAM},project={PROJECT}",
+        ]
+
+        run_gcloud_cmd(cmd)
+
+        steps.update(25)
+
+
+def render_connector_configuration(file_path):
+    """
+    Reads the connector configuration file and replaces all the values that start with `vault:`
+    with the values from Vault
+    """
+    configuration = {}
+    with open(file_path, "r") as f:
+        configuration = json.loads(f.read())
+        for key, item in configuration.items():
+            if type(item) is str and item.startswith("vault:"):
+                configuration[key] = read_from_vault(item)
```

Comment:
Would be nice to have error handling here to include cases like file is not there, no access to file, no valid json is there, there are no `vault:` keys.

Also which keys can be there? Why that `vault:` prefix check?

Reply:
Ah I see - you open connector config and some values are `vault: <key>` and then you replace these with values from the vault. Amazing!

How are 403 messages shown for these? Just JSON of vault response?

Reply:
Yes, if there is an error it will be propagated from Vault.  Like: 

```
No value found at secret/unexisted_secret
```